packages/mui-base/src/useMenuItem/useMenuItem.test.tsx

```diff
@@ -54,5 +64,73 @@ describe('useMenuItem', () => {
       fireEvent.click(menuItem);
       expect(handleClick.callCount).to.equal(1);
     });
+
+    it('should close menu when menu item is clicked', () => {
+      const dispatchSpy = spy();
+      const context = {
+        ...testContext,
+        dispatch: dispatchSpy,
+      };
+
+      function TestMenuItem() {
+        const rootRef = React.createRef<HTMLElement>();
+        const { getRootProps } = useMenuItem({ rootRef, id: 'test-menu-item-1' });
+        return <div {...getRootProps({ 'data-testid': 'test-menu-item' })} />;
+      }
+
+      function Test() {
+        return (
+          <DropdownContext.Provider value={context}>
+            <Menu>
+              <TestMenuItem />
+            </Menu>
+          </DropdownContext.Provider>
+        );
+      }
+
+      render(<Test />);
+
+      const menuItem = screen.getByTestId('test-menu-item');
+      expect(menuItem).not.to.equal(null);
+
+      fireEvent.click(menuItem);
+      expect(dispatchSpy.called).to.equal(true);
```

Comment:
We should check if the actual "close" action is dispatched, not any action. Similarily, below, check if the "close" action is not dispatched.